src/test/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/end2end/AdminPageEnd2EndTest.java

```diff
@@ -0,0 +1,109 @@
+package edu.ucsb.cs56.ucsb_open_lab_scheduler.end2end;
+
+import com.github.tomakehurst.wiremock.extension.responsetemplating.ResponseTemplateTransformer;
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.junit.*;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+
+/**
+ * This test runs a complete end-to-end test on the project as a single test.
+ * Starts by logging in as a student, submitting a review, and reviewing 5 ideas
+ *
+ * If you are trying to duplicate this test in a project, make sure to copy the
+ * html template under the test/resources/__files/ directory!
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT, properties="spring.datasource.name=end2endtest")
+public class AdminPageEnd2EndTest {
+
+    static {
+        System.setProperty("spring.security.oauth2.client.provider.wiremock.authorization-uri", "http://localhost:8077/oauth/authorize");
```

Comment:
I wonder whether we should factor this out into a test helper class support method...  basically everything in this static block...

Reply:
I think the factoring out is optional for merging.. but still a good idea...